docs/src/showcase.md

```diff
@@ -422,6 +422,80 @@ DocTestSetup = nothing
 DocTestSetup = nothing
 ```
 
+### Teardown code
+
+Dually to setup code described in the preceding section it can be useful to have code that
+gets executed *after* the actual doctest, for example to restore a setting or release
+a resource acquired during setup.
+For example, the following doctest expects that `setprecision` was used to
+change the default precision for `BigFloat`. After the test completes, this should
+be returned to the correct setting.
+
+!!! note
+    In real code it usually is better to use `setprecision` with a `do`-block
+    to temporarily change the precion.
+
+```jldoctest; setup=:(oldprec=precision(BigFloat);setprecision(BigFloat,20)), teardown=:(setprecision(BigFloat,oldprec))
```

Comment:
I verified that if I remove the `teardown` here, the doctest fails.

So this constitutes a simple test for the teardown feature.